geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionDataStore.java

```diff
@@ -1472,6 +1472,19 @@ public boolean removeBucket(int bucketId, boolean forceRemovePrimary) {
       }
 
       BucketAdvisor bucketAdvisor = bucketRegion.getBucketAdvisor();
+      InternalDistributedMember primary = bucketAdvisor.getPrimary();
+      InternalDistributedMember myId =
+          this.partitionedRegion.getDistributionManager().getDistributionManagerId();
+      if (primary == null || myId.equals(primary)) {
```

Comment:
The forceRemovePrimary WAS useless and it can be removed because it always use "false".

But when I added the logic to remove the leader region bucket (when the shadow bucket failed to initialize), then I have to call the removeBucket(xxx, true) by myself. 

The shadow bucket was removed first by the exception handling. But I added the logic to remove leader bucket, I have to skip a few "return false" exit points, because at this time the leader bucket is not logically ready and not qualified to be removed unless I force to remove it. 

So I make use of the forceRemovePrimary parameter. Maybe I should change it to better name, such as forceToRemove.